### DISCUSSION TITLE
Networking issues in EU region using self-hosted runners

### DISCUSSION THREAD

**Discussion by knjoroge** [community/community#147304] [Actions]
### Select Topic Area

Bug

### Body

Hello,

We are experiencing networking issues in the EU region with our self-hosted runners on Google Cloud Platform when connecting to GitHub.

The error occurs while fetching our repository and affects both HTTPS and SSH protocols. This issue is happening using self-hosted runners and does not occur when using GitHub-hosted runners.

Are there any networking issues going on GitHub's side? Could this be related?
https://www.githubstatus.com/incidents/d33mtmnttgsh
 
## HTTPS
![image](https://github.com/user-attachments/assets/d0c5ba53-079e-4143-9a07-31d9895d2a97)

## SSH
![image](https://github.com/user-attachments/assets/9412baa4-1841-4ad2-ab32-8c9d01aefa04)

**Comment by alvicsam** [comment]
Hello, I can second this issue. We also use self-hosted runners on Google Cloud Platform and experiencing the same problems.
Example: https://github.com/paritytech/polkadot-sdk/actions/runs/12372199701/job/34529967937

**Reply by usmonster** [reply]
This also affects us and is not limited to our self-hosted runners, but also to any of our services that run Git operations interacting with GitHub from **GCP's EU region**, e.g. Flux (using GitRepository), Atlantis, Jenkins agents on GCE, etc.

The issue may be on Google's side, but we've opened support tickets both to GitHub and to Google.

@alvicsam, are your runners on GCP hosted in the EU region as well?

**Reply by alvicsam** [reply]
@usmonster yes

**Reply by usmonster** [reply]
Most recent response from Google:

> [...] we continue to work with Github's network engineering team to find a traffic-management solution that doesn't affect normal GCP use cases. We don't have an ETA to implement this solution yet.
The traffic management applies primarily to users of Github in Europe, so a workaround is possible if you can use Github from GCP instances outside European regions.

We're constrained to stay within EU regions by performance & legal reasons, so asking them for another option.

**Reply by alvicsam** [reply]
@usmonster thank you for the update 🙏

**Reply by usmonster** [reply]
No details yet, but Google reports they've resolved the network issue on their side. 😌
